Sign in ml

mlimberg opened the pull request:
#### This PR includes the following:
* Ability to sign in
    * Sets active user in store with all compositions and sounds tied to that ID
* Error messages prompted if username and/or password do not match

#### Issue
* Modal does not disappear on blur
* Signing in then clicking logout shows sign in modal again

esayler commented on lib/containers/UserContainer.jsx:
```diff
@@ -1,20 +1,20 @@
 const mapStateToProps = (state) => {
   return {
     user: state.ActiveUser.username,
-    error: state.ActiveUser.error || false
```
the || false could be refactored out?

esayler commented on lib/reducers/ActiveUser.js:
```diff
@@ -2,10 +2,11 @@
-      return { error: action.error }
+      return { error: action.error, type: action.type }
+    case 'NOT_FOUND':
+      return { error: action.error, type: action.type }
```
Could be refactored to a single user error action, such as `USER_ERROR` that holds the error type and error message.

esayler commented on server.js:
```diff
@@ -444,3 +478,15 @@ app.get('*', function (request, response) {
+
+something = 'Have a great day!';
+
+say();
```
remove this from the server.

mlimberg commented:
@esayler @dylanavery720 @DanGrund 

#### Updated:
* Fixed all of Eric's comments
* Imported module to help hide modal on blur (react-onClickOutside)
* Added "X" close button to modal (button no longer toggles it with new blur feature)
* Added some minor styling to modal

esayler commented:
Alright looks good 👍
